packages/api/src/auth/decoders/clerk.ts

```diff
@@ -1,41 +1,30 @@
-import type { APIGatewayProxyEvent, Context as LambdaContext } from 'aws-lambda'
+import type IClerk from '@clerk/clerk-sdk-node/instance'
 
-interface Req {
-  event: APIGatewayProxyEvent
-  context: LambdaContext
-}
-
-export const clerk = async (token: string, req: Req) => {
+export const clerk = async (token: string) => {
   // Use require here, to prevent needing clerk sdk in api deps
-  const { sessions, users } = require('@clerk/clerk-sdk-node')
-
-  if (!process.env.CLERK_API_KEY) {
-    console.error('CLERK_API_KEY env var is not set.')
-    throw new Error('CLERK_API_KEY env var is not set.')
-  }
+  const Clerk = require('@clerk/clerk-sdk-node/instance').default
 
-  // Clerk sessions are a combination of a clerk "current session id", which we store
-  // in the Redwood auth token, and the __session cookie, which contains a second session
-  // bearer token. The two tokens together define which device is browsing and as who.
-  const clerkCookieName = '__session'
-  const cookies = req.event.headers['cookie']?.split(';').map((c) => c.trim())
-  const sessionCookie = cookies
-    ?.find((c) => c.startsWith(clerkCookieName + '='))
-    ?.substring(clerkCookieName.length + 1)
+  const { users, base }: IClerk = new Clerk()
 
-  if (!sessionCookie || sessionCookie.length < 1) {
-    return Promise.reject(new Error('Clerk __session token is not set'))
+  if (!process.env.CLERK_JWT_KEY) {
+    console.error('CLERK_JWT_KEY env var is not set.')
+    throw new Error('CLERK_JWT_KEY env var is not set.')
   }
 
-  const session = await sessions.verifySession(token, sessionCookie)
-  if (!session.userId) {
-    return Promise.reject(new Error('Session invalid'))
-  }
+  try {
+    const jwtPayload = await base.verifySessionToken(token)
+
+    if (!jwtPayload.sub) {
+      return Promise.reject(new Error('Session invalid'))
+    }
 
-  const user = await users.getUser(session.userId)
+    const user = await users.getUser(jwtPayload.sub)
```

Comment:
Will this make an API call to Clerk ti fetch the user per decoding request and is that ok?

Or should we instead update the getCurrentUser template to do that request but .... well, it may still have to re-fetch it as serverless won't necessarily have this in the global context.

Most important really are the `roles`. The rest of the `user` profile info can be done separately.

Reply:
Also, if we cannot for v1 have roles in Clerk, we can skip and users can manage roles on their own -- if that is better?

Reply:
```
  public async getUser(userId: string) {
    this.requireId(userId);
    return this._restClient.makeRequest<User>({
      method: 'GET',
      path: `/users/${userId}`,
    });
  }
```

That's what we're calling. So yeah, that's a network request. But this is no different to what we already have.  Solving this should be another PR.

Reply:
Same for the roles. This PR doesn't change anything with the roles compared to what we already have here.